services/web/server/src/simcore_service_webserver/director_v2_core_dynamic_services.py

```diff
@@ -0,0 +1,218 @@
+) -> None:
+    # stopping a service can take a lot of time
+    # bumping the stop command timeout to 1 hour
+    # this will allow to sava bigger datasets from the services
+    settings: DirectorV2Settings = get_plugin_settings(app)
+    await request_director_v2(
+        app,
+        "DELETE",
+        url=(settings.base_url / f"dynamic_services/{service_uuid}").update_query(
+            save_state="true" if save_state else "false",
+        ),
+        expected_status=web.HTTPNoContent,
+        timeout=settings.DIRECTOR_V2_STOP_SERVICE_TIMEOUT,
+    )
+
+
+@log_decorator(logger=log)
+async def stop_dynamic_services_in_project(
+    app: web.Application,
+    user_id: Optional[PositiveInt] = None,
+    project_id: Optional[str] = None,
+    save_state: bool = True,
+) -> None:
+    """Stops all services of either project_id or user_id in concurrently"""
+    running_dynamic_services = await get_dynamic_services(
+        app, user_id=user_id, project_id=project_id
+    )
+
+    services_to_stop = [
+        stop_dynamic_service(
+            app=app, service_uuid=service["service_uuid"], save_state=save_state
+        )
+        for service in running_dynamic_services
+    ]
+    await logged_gather(*services_to_stop)
+
+
+# FIXME: ANE please unduplicate the 2 following calls!!!!
+@log_decorator(logger=log)
+async def retrieve(
+    app: web.Application, service_uuid: str, port_keys: list[str]
+) -> DataType:
+    """Pulls data from connections to the dynamic service inputs"""
+    settings: DirectorV2Settings = get_plugin_settings(app)
+    result = await request_director_v2(
+        app,
+        "POST",
+        url=settings.base_url / f"dynamic_services/{service_uuid}:retrieve",
+        data={"port_keys": port_keys},
+        timeout=settings.get_service_retrieve_timeout(),
+    )
+    assert isinstance(result, dict)  # nosec
+    return result
+
+
+@log_decorator(logger=log)
+async def request_retrieve_dyn_service(
+    app: web.Application, service_uuid: str, port_keys: list[str]
+) -> None:
+    # TODO: notice that this function is identical to retrieve except that it does NOT reaise
+    settings: DirectorV2Settings = get_plugin_settings(app)
+    body = {"port_keys": port_keys}
+
+    try:
+        await request_director_v2(
+            app,
+            "POST",
+            url=settings.base_url / f"dynamic_services/{service_uuid}:retrieve",
+            data=body,
+            timeout=settings.get_service_retrieve_timeout(),
+        )
+    except DirectorServiceError as exc:
+        log.warning(
+            "Unable to call :retrieve endpoint on service %s, keys: [%s]: error: [%s:%s]",
+            service_uuid,
+            port_keys,
+            exc.status,
+            exc.reason,
+        )
+
+
+@log_decorator(logger=log)
+async def restart_dynamic_service(app: web.Application, node_uuid: str) -> None:
+    settings: DirectorV2Settings = get_plugin_settings(app)
+    await request_director_v2(
+        app,
```

Comment:
FYI: the name is misleading. The only thing this does is to restart the service containers which are already running. Could you please ad a note to change this in the future to a use a better name?